src/Catalog/Monitoring/Resources/PackageTimestampMetadataResourceV3.cs

```
@@ -0,0 +1,59 @@
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Providers
+{
+    public class PackageTimestampMetadataResourceV3 : IPackageTimestampMetadataResource
+    {
+        public PackageTimestampMetadataResourceV3(RegistrationResourceV3 registration)
+        {
+            _registration = registration;
+        }
+        
+        private readonly RegistrationResourceV3 _registration;
+
+        /// <summary>
+        /// Parses the registration for the package specified by the <see cref="ValidationTestRunData"/> and returns a <see cref="PackageTimestampMetadata"/> created from the package's catalog entry.
+        /// If the registration page is missing, search for a deletion catalog entry and use its timestamp.
+        /// </summary>
+        public async Task<PackageTimestampMetadata> Get(ValidationTestRunData data)
+        {
+            var registrationPage = await _registration.GetPackageMetadata(data.Package, NullLogger.Instance, data.CancellationToken);
+
+            if (registrationPage == null)
+            {
+                DateTime? deleted = null;
+
+                var deleteEntries = data.Entries.Where(entry => entry.IsDelete());
+                if (deleteEntries.Any())
+                {
+                    deleted = deleteEntries.Max(entry => entry.CommitTimeStamp);
+                }
+
+                return new PackageTimestampMetadata(deleted);
+            }
+            
+            try
+            {
+                var catalogEntryUri = new Uri(registrationPage.GetValue("catalogEntry").ToString());
+                var catalogPage = await data.Client.GetJObjectAsync(catalogEntryUri);
+
+                var created = catalogPage.GetValue("created").Value<DateTime>();
+                var lastEdited = catalogPage.GetValue("lastEdited").Value<DateTime>();
+
+                return new PackageTimestampMetadata(created, lastEdited);
+            }
+            catch
```

>catch [](start = 12, length = 5)

I don't think we should blindly swallow any exception here